internal/pkg/manifest/validate.go

```diff
@@ -2242,6 +2243,11 @@ func validateAndPopulateNLBListenerPorts(listener NetworkLoadBalancerListener, p
 		targetProtocol = strings.ToUpper(aws.StringValue(nlbProtocol))
 	}
 
+	// Handle TLS termination of container exposed port protocol
+	if targetProtocol == TLS {
+		targetProtocol = TCP
+	}
```

Comment:
I really am not a fan of nested if-block, but I think it's more obvious this way that we set the target protocol to TCP if the _listener protocol_ is TLS, not necessarily if the target protocol is "tls". Imagine if we expand `target_port` to accept a string of form `443/tls` (for example to let people configure e2e encryption), then this code might become slightly confusing to refactor.


```suggestion
		if strings.EqualFold(aws.StringValue(nlbProtocol), "tls") {
		  targetProtocol = TCP
		} else {
		   targetProtocol = strings.ToUpper(aws.StringValue(nlbProtocol))
		}
```